internal/storage/sqlmock_provider.go

```diff
@@ -0,0 +1,60 @@
+package storage
```

Comment:
I don't see where this mock is used. It seems to be dead code. Please either drop it or add a test using it.